### PR TITLE
Unpin click dependency

### PR DESCRIPTION
**Pull request by kiersten-stokes**
### What changes were proposed in this pull request?
Our recent test failures seem to be due to a conflicting pin of `click` (we pin <8 whereas the latest `black` release pins `click` at >= 8). This PR removes the pin on `click` (imposed in #2010) as further research revealed that the KFP `click` dependency is only for use in the KFP CLI.

### How was this pull request tested?
Tests are passing. Also ran some pipelines and all are passing.

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Review by ptitzler — APPROVED** [review]
LGTM

File: setup.py [hunk 1]
```diff
@@ -72,7 +72,7 @@
     packages=find_packages(),
     install_requires=[
         'autopep8>=1.5.0,<1.5.6',
-        'click>=7.1.1,<8',  # Required by kfp 1.6.3
+        'click',
```
Comment (kevin-bates) [inline setup.py:75]: I think we should leave the floor in.  Floors are much safer than caps. :smile:

**Review by kevin-bates — COMMENTED** [review]
The cap on `jupyter_core` (5.0) should be removed as well.  We're close enough to that repo (I personally watch it) to be aware of any breaking major releases.

Other caps that reference non-existent releases are:
- `minio` (8.0) This cap was updated from 7 when a contribution required version 7, implying there was no reason for it to be capped at 7 in the first place.  The contributor added 8, I'm sure, due to precedence.  Btw, the same [PR](https://github.com/elyra-ai/elyra/pull/2335) that moved the `minio` cap to 8.0 also updated the `kfp` base to `1.7.0`, so the comment regarding `click` and `kfp==1.6.3` is moot anyway.
- `pyyaml` (6.0) - This cap looks like it [might be legit](https://github.com/yaml/pyyaml/issues/576) as 6.0 does (now) exist.  However, if we use `yaml.safe_load()` - which appears to be the case, then `safe_load()` calls `load()` with the `SafeLoader` loader - so all is good.
- `requests` (3.0) - Non-existent release.  However, their [docs](https://docs.python-requests.org/en/latest/community/release-process/#release-process-and-rules) indicate that major releases will contain breaking changes.  As a result, I think they view releases differently in that new features that are backward compatible will not be a major release. 

I understand not wanting to remove other caps because as soon as we do, it will probably bite us. But, on the other hand, I think this is the kind of thing that should be due to a real issue, not fear-based.  If folks run into issues due to new major releases, then a) the community is probably having the same issues, and b) it's a matter of downgrading the impacted package. (My two cents)

**Comment by kevin-bates** [comment]
I'm okay only addressing the `click` issue right now.  I would like to see the floor restored since we have to assume there was some reason for it being there in the first place. 

The lack of consistency with caps (or not) is what really bugs me.  It seems completely arbitrary for those cases that use a cap.

**Comment by ptitzler** [comment]
> I'm okay only addressing the `click` issue right now. I would like to see the floor restored since we have to assume there was some reason for it being there in the first place.

I agree. The purpose of this PR should only be to resolve the failing tests. This unblocks other PRs that used to pass validation just fine.

**Comment by kevin-bates** [comment]
I'm going ahead with this merge so that tests will start passing (or not be side-affected by this).

**Pull request merged**